sql-plugin/src/main/spark330db/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastNestedLoopJoinExec.scala

```diff
@@ -222,17 +223,20 @@ case class GpuBroadcastNestedLoopJoinExec(
     }
   }
 
-  override def makeBuiltBatchInternal(
+  override def makeBuiltBatchAndStreamIter(
       relation: Any,
+      streamIter: Iterator[ColumnarBatch],
+      buildSchema: StructType,
       buildTime: GpuMetric,
-      buildDataSize: GpuMetric): ColumnarBatch = {
+      buildDataSize: GpuMetric): (ColumnarBatch, Iterator[ColumnarBatch]) = {
     // NOTE: pattern matching doesn't work here because of type-invariance
     if (isExecutorBroadcast) {
       val rdd = relation.asInstanceOf[RDD[ColumnarBatch]]
-      makeExecutorBuiltBatch(rdd, buildTime, buildDataSize)
+      val builtBatch = makeExecutorBuiltBatch(rdd, buildTime, buildDataSize)
```

Comment:
NIT:
Not related to this PR, but we can make the same optimization for this case too.